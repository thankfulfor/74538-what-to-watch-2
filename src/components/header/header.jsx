import * as React from 'react';
import PropTypes from 'prop-types';
import {Link} from 'react-router-dom';
import {URLS} from '../../utils/constants.js';
import {connect} from 'react-redux';

export const Header = (props) => {
  const {title, isLoggedIn, avatarUrl} = props;

  return (
    <header className="page-header user-page__head">
      <div className="logo">
        <Link to={URLS.MAIN_PAGE_URL} className="logo__link">
          <span className="logo__letter logo__letter--1">W</span>
          <span className="logo__letter logo__letter--2">T</span>
          <span className="logo__letter logo__letter--3">W</span>
        </Link>
      </div>

      {title && <h1 className="page-title user-page__title">{title}</h1>}

      <div className="user-block">
        {isLoggedIn
          ?
          <div className="user-block__avatar">
            <Link to={URLS.MY_LIST_URL}><img src={avatarUrl} alt="User avatar" width="63" height="63" /></Link>
          </div>
          : <Link to={URLS.LOGIN_PAGE_URL}>Sign In</Link>
        }
      </div>
    </header>
  );
};

Header.propTypes = {
  title: PropTypes.string,
  isLoggedIn: PropTypes.bool.isRequired,
  avatarUrl: PropTypes.string.isRequired,
};

const mapStateToProps = (state, ownProps) => {
  return Object.assign({}, ownProps, {
    avatarUrl: URLS.BASE_URL + state.userData.avatar_url,
    isLoggedIn: state.isLoggedIn
  });
};

export default connect(mapStateToProps)(Header);
