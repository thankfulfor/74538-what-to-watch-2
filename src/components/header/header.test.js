import React from 'react';
import {Header} from './header.jsx';
import Enzyme, {shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({adapter: new Adapter()});

it(`Header rendered`, () => {
  const historyMock = {push: jest.fn()};
  const tree = shallow(<Header history={historyMock} avatarUrl="" isLoggedIn={false} />);

  expect(tree).toMatchSnapshot();
});
