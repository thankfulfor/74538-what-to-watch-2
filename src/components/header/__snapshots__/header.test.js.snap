// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header рендерится корректно 1`] = `ShallowWrapper {}`;
